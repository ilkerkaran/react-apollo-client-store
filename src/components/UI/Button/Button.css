.Button {
  background-color: transparent;
  border: none;
  color: white;
  outline: none;
  cursor: pointer;
  font: inherit;
  padding: 10px;
  margin: 10px;
  font-weight: bold;
  text-align: center;
  text-align:center;
  display: inline-block;
}

.Success {
  color: #5c9210;
}

.Danger {
  color: #944317;
}
