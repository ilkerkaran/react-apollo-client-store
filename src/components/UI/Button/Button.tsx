import React from 'react';
import './Button.css';

const Button = ({
  inputType, buttonType, onClick, children
}: { inputType: 'button' | 'reset' | undefined, buttonType: 'Success' | 'Danger' | undefined, onClick: () => any, children: any}) => (
  <button
      // eslint-disable-next-line react/button-has-type
    type={inputType}
    className={['Button', buttonType].join(' ')}
    onClick={onClick}
  >
    {children}
  </button>
);

export default Button;
