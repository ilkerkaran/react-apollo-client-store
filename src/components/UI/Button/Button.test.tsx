import React from 'react';
import renderer from 'react-test-renderer';

import Button from './Button';

describe('<Button /> snapshot ', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Button
        buttonType="Success"
        inputType="button"
        onClick={() => {}}
      />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
