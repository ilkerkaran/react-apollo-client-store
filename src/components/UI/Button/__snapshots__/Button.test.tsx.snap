// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> snapshot  renders correctly 1`] = `
<button
  className="Button Success"
  onClick={[Function]}
  type="button"
/>
`;
